pkg/cmd/server/kubernetes/master_config.go

```diff
@@ -106,7 +93,95 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
 	// TODO: this should be done in config validation (along with the above) so we can provide
 	// proper errors
 	if err := cmdflags.Resolve(options.KubernetesMasterConfig.APIServerArguments, server.AddFlags); len(err) > 0 {
-		return nil, kerrors.NewAggregate(err)
+		return nil, nil, kerrors.NewAggregate(err)
+	}
+
+	resourceEncodingConfig := genericapiserver.NewDefaultResourceEncodingConfig()
+	resourceEncodingConfig.SetVersionEncoding(
```

Comment:
From our old code - this is our encoded historical choices about things we took from kube.